tests/integration_tests/flows/test_clickhouse.py

```diff
@@ -44,92 +56,226 @@ def query_ch(query, database='default'):
     )
 
     if res.status_code != 200:
-        print(f'Error in query: {query}')
+        print(f'ERROR: code={res.status_code} msg={res.text}')
+        raise Exception()
 
     if ' FORMAT JSON' in query:
-        return res.json()['data']
-    else:
-        return res.text
+        res = res.json()['data']
+
+    return res
+
 
 class ClickhouseTest(unittest.TestCase):
     @classmethod
     def setUpClass(cls):
-        set_get_config_path()
-
-        cls.sp = Popen(['python3', '-m', 'mindsdb'], close_fds=True)
-
-        for i in range(20):
-            try:
-                res = requests.get(f'{root}/util/ping')
-                if res.status_code != 200:
-                    raise Exception('')
-            except:
-                time.sleep(1)
-                if i == 19:
-                    raise Exception("Can't connect !")
-
-        #query_ch('DROP DATABASE mindsdb')
-        query_ch(f"""
-        CREATE TABLE {ds_name} (number_of_rooms String, number_of_bathrooms String, sqft Int64, location String, days_on_market Int64, initial_price Int64, neighborhood String, rental_price Float64)  ENGINE=URL('https://raw.githubusercontent.com/mindsdb/mindsdb-examples/master/benchmarks/home_rentals/dataset/train.csv', CSVWithNames)
+        cls.mdb = MindsdbNative(config)
+
+        models = cls.mdb.get_models()
+        models = [x['name'] for x in models]
+        if TEST_PREDICTOR_NAME in models:
+            cls.mdb.delete_model(TEST_PREDICTOR_NAME)
+
+        query('create database if not exists test')
+        test_tables = query('show tables from test')
+        test_tables = [x['name'] for x in test_tables]
+        if TEST_DATA_TABLE not in test_tables:
+            print('creating test data table...')
+            query(f'''
+                CREATE TABLE test.{TEST_DATA_TABLE} (
```

Comment:
Can we switch back t creating the table over the network:

```
CREATE TABLE {ds_name} (number_of_rooms String, number_of_bathrooms String, sqft Int64, location String, days_on_market Int64, initial_price Int64, neighborhood String, rental_price Float64)  ENGINE=URL('https://raw.githubusercontent.com/mindsdb/mindsdb-examples/master/benchmarks/home_rentals/dataset/train.csv', CSVWithNames)
```

makes things much simpler to read

Reply:
agree, file small, we can reload it each time

Reply:
But may be better to keep MergeTree-engine table for tests.

Reply:
after some experiments i conclude what ENGINE=URL('', CSVWithNames) very buggy:  you can specify any types of columns and table will be created. But you can access only rows what can by cast to that types, other will be missed, or it will be error if you try to access them. Example:
1. create table `CREATE TABLE test.xxx3 (number_of_rooms String, number_of_bathrooms String, sqft String, location String,days_on_market Int16,initial_price Int32,neighborhood String,rental_price Int32) ENGINE=URL('https://s3.eu-west-2.amazonaws.com/mindsdb-example-data/home_rentals.csv', CSVWithNames)` 
number_of_rooms, number_of_bathrooms, sqft strings, overwise will be error in any access.
2.`select * from test.xxx3 order by rental_price limit 1;` - error by timeout
3. `select * from test.xxx3 limit 1;` works, but  `select * from test.xxx3 limit 10;` fail by `Expected end of line.`
so i keep old table creation

Reply:
Fair enough. I can't say I like it but I agree on the fact that engine switching is easier this way and if you think the URL impl is buggy since it breaks only when selecting rather than at creation... than, yeah, I see your point.